video/out/w32_common.c

```diff
@@ -605,6 +605,75 @@ static void update_playback_state(struct vo_w32_state *w32)
                                                     TBPF_NORMAL);
 }
 
+struct get_monitor_data {
+    int i;
+    int target;
+    HMONITOR mon;
+};
+
+static BOOL CALLBACK get_monitor_proc(HMONITOR mon, HDC dc, LPRECT r, LPARAM p)
+{
+    struct get_monitor_data *data = (struct get_monitor_data*)p;
+
+    if (data->i == data->target) {
+        data->mon = mon;
+        return FALSE;
+    }
+    data->i++;
+    return TRUE;
+}
+
+static HMONITOR get_monitor(struct vo_w32_state *w32)
```

Comment:
I don't like how get_monitor() has been repurposed here. get_monitor() was a simple stateless utility function to get an HMONITOR by ID. Now it seems like logic from update_screen_rect() has been folded into get_monitor(). This should be in a separate function and get_monitor() should stay simple and not depend on struct vo_w32_state. Or maybe the original get_monitor() should be retained as get_monitor_by_id() and the new get_monitor() could call it.